### PR TITLE
Use kind-projector plugin wherever possible

### PR DESCRIPTION
**Pull request by stephenjudkins**
I find this a lot easier to read, and it's how we're doing it everywhere else

### CODE REVIEW

File: core/src/main/scala/cats/Apply.scala [hunk 1]
```diff
@@ -39,7 +39,7 @@ trait Apply[F[_]] extends Functor[F] with ApplyArityFunctions[F] { self =>
    * val y = Some(List(10, 20))
    * ap.map2(x, y)(_ + _) == Some(List(11, 12, 21, 22))
    */
-  def compose[G[_]](implicit GG: Apply[G]): Apply[({type λ[α] = F[G[α]]})#λ] =
+  def compose[G[_]](implicit GG: Apply[G]): Apply[λ[ A => F[G[A]] ]] =
```
Comment (ceedubs) [inline core/src/main/scala/cats/Apply.scala]: Scalastyle doesn't like the space after the`λ[`.

Comment (stephenjudkins) [inline core/src/main/scala/cats/Apply.scala]: That's too bad, I think it makes it easier to read. I am happy to change it, though

**Comment by ceedubs** [comment]
:+1:

**Comment by non** [comment]
:+1:

**Comment by refried** [comment]
I haven't been paying close attention; should we stick with upper-case type variables in the type lambdas (unlike scalaz), or is it unspecified?

**Pull request merged**